common-ui/src/main/java/com/duckduckgo/app/global/BaseDuckDuckGoActivity.kt

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) 2022 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.global
+
+import android.annotation.SuppressLint
+import android.content.BroadcastReceiver
+import android.os.Bundle
+import android.view.MenuItem
+import androidx.appcompat.widget.Toolbar
+import androidx.localbroadcastmanager.content.LocalBroadcastManager
+import com.duckduckgo.mobile.android.R
+import com.duckduckgo.mobile.android.ui.applyTheme
+import com.duckduckgo.mobile.android.ui.store.ThemingDataStore
+import dagger.android.AndroidInjection
+import dagger.android.DaggerActivity
+import javax.inject.Inject
+
+abstract class BaseDuckDuckGoActivity : DaggerActivity() {
```

Comment:
This could probably be named better but I am out of ideas + I don't want to change DuckDuckGoActivity's name.